test/unit/node_shared_test.rb

```diff
@@ -20,6 +20,32 @@ class NodeSharedTest < ActiveSupport::TestCase
     assert_equal 5, html.scan('notes-grid-test').length
     assert html.scan('<td class="title">').length > 1
   end
+ 
+  test 'that NodeShared can be used to convert short codes like [button:foo:https://google.com] into tables which list buttons' do
+    before = "Here are some notes in a table: \n\n[button:Press me:/questions] \n\n[button:Cancel:https://google.com]\n\n`[button:Cancel:https://google.com]` This shouldn't get recognized because it's in ` ticks.\n\nMake sense?"
+    html = NodeShared.button(before)
+    assert html
+    assert_equal 1, html.scan('<table class="table inline-grid notes-grid notes-grid-test notes-grid-test-').length
+    assert_equal 1, html.scan('<table').length
+    assert_equal 5, html.scan('button-grid-test').length
```

Comment:
@jywarren is the number 5 here correct? I'm not sure if all of this is only a length of 5 but I could be wrong

Reply:
You're doing really well here, awesome!

Here, lines 24-7 are perfect. But starting on line 28-30, you seem to be searching the string for a table, which is what other shortcodes produce. Your shortcode should produce button HTML, in the format `<a class="btn btn-primary"...`, right? so you should scan for things like `btn btn-primary` and `href=`... does that make sense?

Reply:
@jywarren thanks! I understand, makes more sense than a table :)